salt/modules/swarm.py

```diff
@@ -0,0 +1,360 @@
+# -*- coding: utf-8 -*-
+"""
+Docker Swarm Module using Docker's Python SDK
+=============================================
+
+:codeauthor: Tyler Jones <jonestyler806@gmail.com>
+
+.. versionadded:: 2017.7.2
+
+The Docker Swarm Module is used to manage and create Docker Swarms.
+
+Dependencies
+============
+
+- Docker installed on the host
+- Docker python sdk >= 2.5.1
+
+Docker Python SDK
+=================
+pip install -U docker
+
+More information: https://docker-py.readthedocs.io/en/stable/
+"""
+from __future__ import absolute_import
+import docker
+import salt.config
+import salt.loader
+import json
+
+
+def __init__():
+    __context__['client'] = docker.from_env()
+    __context__['server_name'] = __grains__['id']
+
+
+def swarm_tokens():
+    '''
+    Get the Docker Swarm Manager or Worker join tokens
+
+    CLI Example:
+
+    .. code-block:: bash
+
+      salt '*' swarm.swarm_tokens 
+    '''
+    client = docker.APIClient(base_url='unix://var/run/docker.sock')
+    service = client.inspect_swarm()
+    return service['JoinTokens']
+
+
+def swarm_init(advertise_addr=str,
+               listen_addr=int,
+               force_new_cluster=bool):
+    '''
+    Initalize Docker on Minion as a Swarm Manager
+
+    advertise_addr = the ip of the manager
```

Comment:
This is better, but you should format them for rst

```
    advertise_addr

        The ip of the swarm manager
```

Reply:
Got it :) newbie here

Reply:
no worries :+1: just making sure that it is all rendered correctly before it gets merged.

If you want to build the docs your self and test the builds locally, check out https://docs.saltstack.com/en/latest/topics/development/conventions/documentation.html
and
https://docs.saltstack.com/en/latest/topics/development/conventions/documentation.html#docs-building

Reply:
@gtmanfred I think i got most of this fixed unless I missed somthing and I also added the *.rst file for the docs. Let me know if something seems off and thanks again!